src/NuGet.Core/NuGet.Protocol.Core.v3/CachingSourceProvider.cs

```diff
@@ -21,8 +21,8 @@ public class CachingSourceProvider : ISourceRepositoryProvider
         private readonly List<SourceRepository> _repositories = new List<SourceRepository>();
 
         // There should only be one instance of the source repository for each package source.
-        private readonly ConcurrentDictionary<PackageSource, SourceRepository> _cachedSources
-            = new ConcurrentDictionary<PackageSource, SourceRepository>();
+        private readonly ConcurrentDictionary<string, SourceRepository> _cachedSources
+            = new ConcurrentDictionary<string, SourceRepository>(StringComparer.OrdinalIgnoreCase);
```

Comment:
For this OrdinalIgnoreCase, it's matching what the PackageSource was doing when it was the key for this dictionary.
PackageSource.Equals uses OrdinalIgnoreCase.
The RestoreArgs code probably needs to use OrdinalIgnoreCase, but I need to verify that.

---
In reply to: [62937395](https://github.com/NuGet/NuGet.Client/pull/586#discussion_r62937395) [](ancestors = 62937395)